ocaml/xapi/extauth_plugin_ADwinbind.ml

```diff
@@ -81,6 +89,21 @@ let ntlm_auth uname passwd : (unit, exn) result =
     Ok ()
   with _ -> Error (auth_ex uname)
 
+let get_domain_info_from_db () =
+  (fun __context ->
+    let host = Helpers.get_localhost ~__context in
+    let service_name =
+      Db.Host.get_external_auth_service_name ~__context ~self:host
+    in
+    let workgroup, netbios_name =
+      Db.Host.get_external_auth_configuration ~__context ~self:host |> fun l ->
+      (List.assoc_opt "workgroup" l, List.assoc_opt "netbios_name" l)
+    in
+    {service_name; workgroup; netbios_name}
+    )
+  |> Server_helpers.exec_with_new_task
```

Comment:
This could be at the beginning of the function like this:
```ocaml
Server_helpers.exec_with_new_task "retrieving external auth domain workgroup" @@ fun __context ->
...
```

Reply:
yes, I see, 
There are places use it and I like to just keep it for consistency